experiments/compositions/composition/expanders/jinja2/main.go

```diff
@@ -0,0 +1,169 @@
+		Manifests: []byte{},
+		Values:    []byte{},
+	}
+
+	log.Printf("EvaluateRequest:\n  config: %s\n  context: %s\n  facade: %s\n  values: %s\n", req.Config, req.Context, req.Facade, req.Value)
+	dir, err := os.MkdirTemp("", "jinja2")
+	if err != nil {
+		newerr := fmt.Errorf("Unexpected tmp file creation failure. %v", err)
+		log.Print(newerr.Error())
+		return nil, newerr
+	}
+	defer os.RemoveAll(dir) // clean up
+
+	e := expander{path: dir, req: req}
+	err = e.WriteInputsToFileSystem()
+	if err != nil {
+		newerr := fmt.Errorf("Error processing inputs: %v", err)
+		log.Print(newerr.Error())
+		return nil, newerr
+	}
+
+	return e.Evaluate(result)
+}
+
+func (e *expander) WriteInputsToFileSystem() error {
+	context := map[string]interface{}{}
+	if string(e.req.Context) != "" {
+		err := json.Unmarshal(e.req.Context, &context)
+		if err != nil {
+			return fmt.Errorf("Error unmarshalling req.Context: %v", err)
+		}
+	}
+
+	facade := map[string]interface{}{}
+	err := json.Unmarshal(e.req.Facade, &facade)
+	if err != nil {
+		return fmt.Errorf("Error unmarshalling req.Facade: %v", err)
+	}
+
+	getterValues := map[string]interface{}{}
+	if string(e.req.Value) != "" {
+		err := json.Unmarshal(e.req.Value, &getterValues)
+		if err != nil {
+			return fmt.Errorf("Error unmarshalling req.Values: %v", err)
+		}
+	}
+
+	valuesObj := map[string]interface{}{
+		"context":      context,
+		e.req.Resource: facade,
+		"values":       getterValues,
+	}
+
+	// marshall values
+	values, err := json.Marshal(valuesObj)
+	if err != nil {
+		return fmt.Errorf("failed to marshal values: %v", err)
+	}
+
+	// Write values to file
+	err = atomicfile.WriteFile(e.path+"/values", values, 0644)
+	if err != nil {
+		return fmt.Errorf("failed to write req.[context,facade,values] to file: %v", err)
+	}
+
+	// Write template to file
+	err = atomicfile.WriteFile(e.path+"/template", e.req.Config, 0644)
+	if err != nil {
+		return fmt.Errorf("failed to write req.Config to file: %v", err)
+	}
+
+	return nil
+}
+
+func (e *expander) Evaluate(result *pb.EvaluateResult) (*pb.EvaluateResult, error) {
+	// wait for /expanded/expanded to be created and then read it
+	args := []string{e.path + "/template", e.path + "/values", "--format=json", "-o", e.path + "/expanded"}
+
+	op, err := exec.Command("jinja2", args...).CombinedOutput()
+	if err != nil {
+		errstring := fmt.Sprintf("Failed evaluating template:\n %s", string(op))
+		log.Print(errstring)
+		result.Errors = append(result.Errors, errstring)
+		result.Status = pb.Status_EVALUATE_FAILED
+		return result, nil
+	}
+
+	manifests, err := os.ReadFile(e.path + "/expanded")
+	if err != nil {
+		newerr := fmt.Errorf("Failed reading expanded file: %v", err)
+		log.Print(newerr)
+		return nil, newerr
+	}
+
+	result.Manifests = manifests
+
+	log.Printf("Result:\n  type: %s\n  error: %s\n  status: %s\n  manifests: %s\n  values: %s\n", result.Type, result.Errors, result.Status, result.Manifests, result.Values)
```

Comment:
Blindly logging values has me worried. For now lets not until we have a solution we like for masking out PII.

Reply:
commented out the logs.